### ISSUE TITLE
Bug: Switching Buffer Between Windows

### ISSUE THREAD

**Issue by jmercouris** [atlas-engineer/nyxt#978]
**Describe the bug**
```
<ERROR> [13:55:43] In #<FUNCTION (LAMBDA (BUFFER) :IN SWITCH-BUFFER) {2228599B}>: There is no applicable method for the generic function
                                                                                    #<STANDARD-GENERIC-FUNCTION NYXT:FFI-WINDOW-SET-ACTIVE-BUFFER (1)>
                                                                                  when called with arguments
                                                                                    (#<USER-WINDOW {100A003903}>
                                                                                     #<BUFFER {1007DE21A3}>).
```
The type signature is interferring with switching the buffer. The buffer could be of ANY class, so we cannot reasonably define a type here.

**Comment by Ambrevar** [comment]
My guess is that the issue is in `window-set-active-buffer`: we create a temp-buffer which is of class `buffer` and not `user-buffer`, thus it does not inherit from gtk-buffer which is expected by `ffi-window-set-active-buffer`.  Does that make sense?

If this verifies, then the fix is simple: replace `(make-instance 'buffer)` with `(make-instance 'user-buffer)`.

**Comment by Ambrevar** [comment]
I guess the bug is only triggered when there is another window already displaying the queried buffer.

**Comment by Ambrevar** [comment]
Fixed in 00cfbe0ae665f13259865d2b5a8d6f1ec6ece860.
